pkg/apis/kubeone/types.go

```diff
@@ -583,6 +620,9 @@ type ProviderStaticNetworkConfig struct {
 
 	// DNS
 	DNS DNSConfig `json:"dns"`
+
+	// IPFamily
+	IPFamily IPFamily `json:"ipFamily"`
```

Comment:
`ProviderStaticNetworkConfig` is used only for worker nodes, so I think we should move this field somewhere else because it's also used in other places, such as for addons and kubeadm config. I think moving it to `ClusterNetworkConfig` would be ideal.